linera-core/src/worker.rs

```diff
@@ -1055,6 +1067,9 @@ where
         notify_when_messages_are_delivered: Option<oneshot::Sender<()>>,
     ) -> Result<(ChainInfoResponse, NetworkActions), WorkerError> {
         trace!("{} <-- {:?}", self.nickname, certificate);
+        NUM_ROUNDS_IN_CERTIFICATE
+            .with_label_values(&[certificate.value().to_log_str()])
+            .observe(certificate.round.0 as f64);
```

Comment:
I would put that at the end of the function, so invalid certificates aren't counted.

And even then, we might want to not count a [duplicate certificate](https://github.com/linera-io/linera-protocol/blob/af233c23c5aa1f9e7a1af9ff45a7f4ed3e1ecc25/linera-core/src/worker.rs#L686).

Reply:
Should I not count a duplicate certificate then? I have to fetch the block and the chain to check if it's a duplicate like in the code you linked, right? Will wait on your answer to address the duplicate part, but will address the rest now

Reply:
Why not just return the information about whether it's a duplicate from `process_validated_block`?